homeassistant/components/romy/vacuum.py

```diff
@@ -0,0 +1,212 @@
+    romy: RomyRobot = coordinator.romy
+
+    device_info = {
+        "manufacturer": "ROMY",
+        "model": romy.model,
+        "sw_version": romy.firmware,
+        "identifiers": {"serial": romy.unique_id},
+    }
+
+    romy_vacuum_entity = RomyVacuumEntity(coordinator, romy, device_info)
+    entities = [romy_vacuum_entity]
+    async_add_entities(entities, True)
+
+
+class RomyVacuumEntity(CoordinatorEntity[RomyVacuumCoordinator], StateVacuumEntity):
+    """Representation of a ROMY vacuum cleaner robot."""
+
+    def __init__(
+        self,
+        coordinator: RomyVacuumCoordinator,
+        romy: RomyRobot,
+        device_info: dict[str, Any],
+    ) -> None:
+        """Initialize the ROMY Robot."""
+        super().__init__(coordinator)
+        self.romy = romy
+        self._device_info = device_info
+        self._attr_unique_id = self.romy.unique_id
+        self._state_attrs: dict[str, Any] = {}
+
+        self._is_on = False
+        self._fan_speed = FAN_SPEEDS.index(FAN_SPEED_NONE)
+        self._fan_speed_update = False
+
+    @property
+    def supported_features(self) -> VacuumEntityFeature:
+        """Flag vacuum cleaner robot features that are supported."""
+        return SUPPORT_ROMY_ROBOT
+
+    @property
+    def fan_speed(self) -> str:
+        """Return the current fan speed of the vacuum cleaner."""
+        return FAN_SPEEDS[self.romy.fan_speed]
+
+    @property
+    def fan_speed_list(self) -> list[str]:
+        """Get the list of available fan speed steps of the vacuum cleaner."""
+        return FAN_SPEEDS
+
+    @property
+    def battery_level(self) -> None | int:
+        """Return the battery level of the vacuum cleaner."""
+        return self.romy.battery_level
+
+    @property
+    def state(self) -> None | str:
+        """Return the state/status of the vacuum cleaner."""
+        return self.romy.status
+
+    @property
+    def is_on(self) -> bool:
+        """Return True if entity is on."""
+        return self._is_on
+
+    @property
+    def name(self) -> str:
+        """Return the name of the device."""
+        return self.romy.name
+
+    @property
+    def icon(self) -> str:
+        """Return the icon to use for device."""
+        return ICON
+
+    @property
+    def device_state_attributes(self) -> Mapping[str, Any] | None:
+        """Return the state attributes of the device."""
+        return self._state_attrs
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn the vacuum on."""
+        LOGGER.debug("async_turn_on")
+        ret = await self.romy.async_clean_start_or_continue()
+        if ret:
+            self._is_on = True
+
+    async def async_return_to_base(self, **kwargs: Any) -> None:
+        """Return vacuum back to base."""
+        LOGGER.debug("async_return_to_base")
+        ret = await self.romy.async_return_to_base()
+        if ret:
+            self._is_on = False
+
+    # turn off robot (-> sending back to docking station)
```

Comment:
We don't need this both in the docstring and as a comment

Reply:
removed the comments